FaunaDB.Client/Types/Value.cs

```diff
@@ -81,7 +81,35 @@ public abstract class Value : Expr
             Field.At(indexes).Get(this).Match(
                 Success: value => value,
                 Failure: reason => NullV.Instance
-                );
+            );
+
+
+        /// <summary>
+        /// Attempts to coerce this value to given type T specified.
+        /// <para>
+        /// See <see cref="Decoder"/>
+        /// </para>
+        /// </summary>
+        /// <exception cref="System.InvalidOperationException">
+        /// Thrown if this instance of <see cref="Value"/> is not convertible to T.
+        /// Or if type T does not fullfil the requirements to be decoded.
+        /// </exception>
+        public T To<T>() =>
```

Comment:
I think there will be always the case where you wan't to traverse the result tree and later on decode it's leaf. For example: 

```csharp
res.Get("key1").Get("key2").To<Product>()
```

In that case, `IResult` comes at hand. Maybe `To` is the final call so, you don't need to return a `IResult` but, at the same time, `IResult` is useful when you are schemaless since you try to decode the value to different types based on their failure instead of `try/catch` conversion attempts.